### PR TITLE
Adds DomainEventHandler a priority property to control execution order

### PR DESCRIPTION
**Pull request by Sherpard**
DomainEventHandlers execution order now can be defined by using @Priority annotation.
If annotation is not present, default implementation assigns Integer.MinValue to priority

Also adds an SPI to alter the behavior of how Event handlers are managed

### CODE REVIEW

**Comment by adrienlauer** [comment]
Thank you for the nice work !

What do you think about using a`@Priority` annotation on the `DomainEventHandler`s to order them instead of an explicit method in the interface ? This annotation is the JSR standard one and is already used in various places of the framework where absolute ordering among classes is necessary.

**Comment by Sherpard** [comment]
That was my first approach, not with standard means, but by adding a priority property @ DomainEventHandler.

Anyway, due to the nature of Events, there could be a need for different logics when handling different events, (Thinking on child events and so on), Maybe we can make a @Priority interceptor the default implementation,

**Comment by Sherpard** [comment]
> Thank you for the nice work !
> 
> What do you think about using a`@Priority` annotation on the `DomainEventHandler`s to order them instead of an explicit method in the interface ? This annotation is the JSR standard one and is already used in various places of the framework where absolute ordering among classes is necessary.

I've updated the PR to use @Priority instead of getPriority() property from DomainEventHandler Inteface :)  way cleaner

File: core/src/test/java/org/seedstack/business/fixtures/event/interceptors/DefaultPriorityEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Copyright © 2013-2019, The SeedStack authors <http://seedstack.org>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.seedstack.business.fixtures.event.interceptors;
+
+import org.seedstack.business.domain.DomainEventHandler;
+
```
Comment (ejemba) [inline core/src/test/java/org/seedstack/business/fixtures/event/interceptors/DefaultPriorityEventHandler.java:11]: I would add javadoc explaining the role of this class.

File: specs/src/main/java/org/seedstack/business/domain/DomainEventInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.seedstack.business.domain;
+
+import java.util.Collection;
+import java.util.List;
+
+
+@SuppressWarnings("rawtypes")
+public interface DomainEventInterceptor {
```
Comment (ejemba) [inline specs/src/main/java/org/seedstack/business/domain/DomainEventInterceptor.java:24]: I would add javadoc explaining the role of this interface.

File: core/src/test/java/org/seedstack/business/fixtures/event/interceptors/DefaultPriorityEventHandler.java (hunk 1 above)
Comment (Sherpard) [reply core/src/test/java/org/seedstack/business/fixtures/event/interceptors/DefaultPriorityEventHandler.java:11]: This is a test case, but already added some javadoc :)

File: specs/src/main/java/org/seedstack/business/domain/DomainEventInterceptor.java (hunk 2 above)
Comment (Sherpard) [reply specs/src/main/java/org/seedstack/business/domain/DomainEventInterceptor.java:24]: added javadoc 👍

**Review by adrienlauer — CHANGES_REQUESTED** [review]
Please reflect the latest changes in the changelog.

**Comment by Sherpard** [comment]
> Please reflect the latest changes in the changelog.

Already updated the changelog, :)

**Comment by Sherpard** [comment]
Already merged on #108

**Comment by adrienlauer** [comment]
Merged through #108

**Pull request closed without merging**